extensions/email/src/main/java/org/mule/extension/email/internal/mailbox/MailboxConnection.java

```diff
@@ -169,4 +183,24 @@ public ConnectionValidationResult validate() {
   private boolean isCurrentFolder(String mailBoxFolder) {
     return folder.getName().equalsIgnoreCase(mailBoxFolder);
   }
+
+  private void handleIllegalArgumentException(String format, IllegalArgumentException e) throws EmailConnectionException {
+    if (e.getMessage().contains(PORT_OUT_OF_RANGE)) {
+      throw new EmailConnectionException(format, e, UNKNOWN_HOST);
+    }
+    throw new EmailConnectionException(format, e);
+  }
+
+  private void handleEmailMessagingException(String format, MessagingException e) throws EmailConnectionException {
+    if (e.getCause() instanceof SocketTimeoutException) {
+      throw new EmailConnectionException(format, e, CONNECTION_TIMEOUT);
+    }
+    if (e.getCause() instanceof ConnectException) {
+      throw new EmailConnectionException(format, e, UNKNOWN_HOST);
```

Comment:
Either UNKNOWN_HOST is the wrong type or this if should be collapsed with the one below